### PR TITLE
improve test coverage

### PR DESCRIPTION
**Pull request by millken**

### CODE REVIEW

File: action/envelope_test.go [hunk 1]
```diff
@@ -29,7 +31,13 @@ func TestEnvelope_Basic(t *testing.T) {
 	tsf2, ok := evlp.Action().(*Transfer)
 	req.True(ok)
 	req.Equal(tsf, tsf2)
+
+	evlp.SetNonce(nonce)
+	req.Equal(evlp.Nonce(), nonce)
```
Comment (Liuhaai) [inline action/envelope_test.go:36]: (expected, actual)

File: action/envelope_test.go [hunk 2]
```diff
@@ -54,6 +62,78 @@ func TestEnvelope_Proto(t *testing.T) {
...
+	for _, test := range tests {
+		bd := &EnvelopeBuilder{}
+		elp := bd.SetNonce(1).
+			SetAction(test).
+			SetGasLimit(100000).
+			Build()
+		evlp, ok := elp.(*envelope)
+		require.True(ok)
+		err = evlp.LoadProto(evlp.Proto())
+
```
Comment (Liuhaai) [inline action/envelope_test.go:131]: add value checking (`require.Equal()`) for data fields in envelope struct

File: action/sealedenvelope_test.go [hunk 3]
```diff
@@ -32,6 +35,77 @@ func TestSealedEnvelope_Basic(t *testing.T) {
...
+	}
+
+	for _, test := range tests {
+		bd := &EnvelopeBuilder{}
+		elp := bd.SetNonce(1).
+			SetAction(test).
+			SetGasLimit(100000).Build()
+		selp := FakeSeal(elp, identityset.PrivateKey(27).PublicKey())
+		_, err = actionToRLP(selp.Action())
+
```
Comment (Liuhaai) [inline action/sealedenvelope_test.go:106]: same

File: action/envelope_test.go [hunk 4]
```diff
@@ -54,6 +62,82 @@ func TestEnvelope_Proto(t *testing.T) {
...
+		require.NoError(err)
+		require.Equal(elp.Version(), evlp.Version())
```
Comment (Liuhaai) [inline action/envelope_test.go:133]: The input data in https://github.com/iotexproject/iotex-core/blob/6b9ce931385dcd188de07a45ba989c3ac5430c3b/action/envelope_test.go#L70-L104 should be checked

File: action/sealedenvelope_test.go [hunk 5]
```diff
@@ -32,6 +35,81 @@ func TestSealedEnvelope_Basic(t *testing.T) {
...
+		bd := &EnvelopeBuilder{}
+		elp := bd.SetNonce(1).
+			SetAction(test).
+			SetGasLimit(100000).Build()
+		selp := FakeSeal(elp, identityset.PrivateKey(27).PublicKey())
+		rlp, err := actionToRLP(selp.Action())
+
+		require.NoError(err)
+
+		require.Equal(elp.Nonce(), rlp.Nonce())
```
Comment (Liuhaai) [inline action/sealedenvelope_test.go:109]: same

File: action/envelope_test.go (hunk 4 above)
Comment (millken) [reply action/envelope_test.go:133]: I don't think it needs to be checked, just to improve test coverage.

https://app.codecov.io/gh/iotexproject/iotex-core/compare/2929/changes

Comment (Liuhaai) [reply action/envelope_test.go:133]: ok

**Review by Liuhaai — APPROVED** [review]

File: action/envelope_test.go (hunk 4 above)
Comment (CoderZhi) [reply action/envelope_test.go:133]: let's add it in another PR

**Pull request merged**